pkg/integrations/prometheus/promql/parser.go

```diff
@@ -0,0 +1,564 @@
+		mQuery.SelectAllSeries = false
+	} else {
+		mQuery.SelectAllSeries = true
+	}
+
+	return mQueryReqs, nil
+}
+
+func handleBinaryExpr(expr *parser.BinaryExpr, mQueryReqs []*structs.MetricsQueryRequest,
+	queryArithmetic []*structs.QueryArithmetic) ([]*structs.MetricsQueryRequest, []*structs.QueryArithmetic, error) {
+
+	var err error
+
+	myid := mQueryReqs[0].MetricsQuery.OrgId
+	timeRange := mQueryReqs[0].TimeRange
+
+	arithmeticOperation := structs.QueryArithmetic{}
+	var lhsRequest, rhsRequest []*structs.MetricsQueryRequest
+	var lhsQueryArth, rhsQueryArth []*structs.QueryArithmetic
+
+	if constant, ok := expr.LHS.(*parser.NumberLiteral); ok {
+		arithmeticOperation.ConstantOp = true
+		arithmeticOperation.Constant = constant.Val
+	} else {
+		lhsRequest, _, lhsQueryArth, err = parsePromQLQuery(expr.LHS.String(), timeRange.StartEpochSec, timeRange.EndEpochSec, myid)
+		if err != nil {
+			return mQueryReqs, queryArithmetic, err
+		}
+		arithmeticOperation.LHS = lhsRequest[0].MetricsQuery.HashedMName
+		queryArithmetic = append(queryArithmetic, lhsQueryArth...)
+	}
+
+	if constant, ok := expr.RHS.(*parser.NumberLiteral); ok {
+		arithmeticOperation.ConstantOp = true
+		arithmeticOperation.Constant = constant.Val
+	} else {
+		rhsRequest, _, rhsQueryArth, err = parsePromQLQuery(expr.RHS.String(), timeRange.StartEpochSec, timeRange.EndEpochSec, myid)
+		if err != nil {
+			return mQueryReqs, queryArithmetic, err
+		}
+		arithmeticOperation.RHS = rhsRequest[0].MetricsQuery.HashedMName
+		queryArithmetic = append(queryArithmetic, rhsQueryArth...)
+	}
+	arithmeticOperation.Operation = getArithmeticOperation(expr.Op)
+	queryArithmetic = append(queryArithmetic, &arithmeticOperation)
+
+	if mQueryReqs[0].MetricsQuery.MQueryAggs == nil {
+		mQueryReqs = lhsRequest
+	} else if mQueryReqs[0].MetricsQuery.HashedMName == uint64(0) {
+		// This means there is a common group by on multiple metrics separated by operators.
+		// So, we need to append the aggregations to each of the Request in LHS and RHS.
+		// And we can discard the current aggregation in the mQueryReqs[0]
+		rhsRequest = appendMetricAggsToTheMQuery(rhsRequest, &mQueryReqs[0].MetricsQuery)
+		mQueryReqs = appendMetricAggsToTheMQuery(lhsRequest, &mQueryReqs[0].MetricsQuery)
+	} else {
+		mQueryReqs = append(mQueryReqs, lhsRequest...)
+	}
+	mQueryReqs = append(mQueryReqs, rhsRequest...)
+
+	return mQueryReqs, queryArithmetic, nil
+}
+
+func appendMetricAggsToTheMQuery(mQueryReqs []*structs.MetricsQueryRequest, mQuery *structs.MetricsQuery) []*structs.MetricsQueryRequest {
+	for _, mQueryReq := range mQueryReqs {
+		currentAggs := mQueryReq.MetricsQuery.MQueryAggs
+		for currentAggs.Next != nil {
+			currentAggs = currentAggs.Next
+		}
+		currentAggs.Next = mQuery.MQueryAggs
+
+		mQueryReq.MetricsQuery.Groupby = mQuery.Groupby
+		mQueryReq.MetricsQuery.SelectAllSeries = mQuery.SelectAllSeries
+		mQueryReq.MetricsQuery.TagsFilters = mQuery.TagsFilters
+	}
+	return mQueryReqs
+}
```

Comment:
Correct me if I'm wrong, but I think this function name is misleading. It sounds like it would update `mQuery` with aggs from `mQueryReqs`. But the code looks like it appends the `mQuery` aggs to the aggs of each element of `mQueryReqs`

Reply:
ahh yes, it appends the `MQuery` to each of the `mQueryReqs`.
I will change the function name to `appendMetricAggsToMQueryRequests()`?

Reply:
Sure, that's a better name